### PR TITLE
Validation model Rspec tests

### PR DESCRIPTION
**Pull request by ZabihullahNooriWardak**
- Validate the attribute of User model.
- Validate the attribute of Post model.
- Adding unit tests for all models.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    it 'returns the most recent 3 posts' do
+      # Create some test posts for the user with different creation times
+      user.posts.create(title: 'Post 1', created_at: 3.days.ago)
+      post2 = user.posts.create(title: 'Post 2', created_at: 2.days.ago)
+      post3 = user.posts.create(title: 'Post 3', created_at: 1.day.ago)
+      post4 = user.posts.create(title: 'Post 4', created_at: Time.now)
+
+      recent_posts = user.recent_posts
+
+      expect(recent_posts).to eq([post4, post3, post2])
```
Comment (Strangeal) [inline spec/models/user_spec.rb:27]: - [ ] 👍 Great progress on incorporating tests for your models! However, I observed that currently, all the model tests are failing. 🚨 I recommend taking a closer look at these failures and making the necessary adjustments to ensure that all tests pass successfully.

### See below for more details:
Screenshot 2024-01-14 at 01 10 11


Happy coding! 😊✅🚀

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @ZabihullahNooriWardak, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Nice *UI* design ✅ 
- Professional `README` template ✅ 
- Correct `GitHub` workflow ✅ 
- Meaningful **Github** commits ✅ 
- Correct `.github/workflows` for linters ✅ 
- Followed *best-practices* ✅ 
- Well structured project ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    it { should belong_to(:user) }
+    it { should belong_to(:post) }
+  end
+
+  describe 'callbacks' do
+    it 'increments comments_counter on post after create' do
+      user = User.create(name: 'Test User')
+      post = Post.create(title: 'Test Post')
+      comment = Comment.create(user:, post:, text: 'Test Comment')
+
```
Comment (Olamarx) [inline spec/models/comment_spec.rb:14]: - [ ] All the tests are still failing as noticed by the previous reviewer, I will advise you ensure they all pass before making submission. 
image

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @ZabihullahNooriWardak, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Linters passed ✔️
- Test written ✔️
- Good use of gitflow ✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Olamarx — APPROVED** [review]
Hi @ZabihullahNooriWardak, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- All test cases passed ✔️
- Readme file professional ✔️
- Linters passed ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**